app/src/auth.py

```diff
@@ -57,18 +65,23 @@ def register():
 
     if request.method == "POST":
 
+        role = request.form.get("role") # get the position of the role switch at time of submission
+        print("Role selected:", role)
         name = request.form.get("name")
         username = request.form.get("username")
         date_of_birth = request.form.get("date_of_birth")
-        grade = getattr(GradeEnum, request.form.get("grade"))   # Parses string into GradeEnum field and assigns proper grade
         email = request.form.get("email")
         confirm_email = request.form.get("confirm_email")
         password = request.form.get("password")
         confirm_password = request.form.get("confirm_password")
 
         age = calculate_age(date_of_birth)
-        user_email = User.query.filter_by(email=email).first()
-        user_name = User.query.filter_by(username=username).first()
+        if role == "teacher":
+            user_email = Teacher.query.filter_by(email=email).first()
+            user_name = Teacher.query.filter_by(username=username).first()
+        else:
+            user_email = User.query.filter_by(email=email).first()
+            user_name = User.query.filter_by(username=username).first()
```

Comment:
There's currently an issue with registration between both roles, where a student and teacher are allowed to register with the same exact email and username, but only the teacher can log into their account—not the student. Check comments below for more details.